dbterd/types.py

```diff
@@ -0,0 +1,31 @@
+from typing import Union
+
+from dbt_artifacts_parser.parsers.catalog.catalog_v1 import CatalogV1
+from dbt_artifacts_parser.parsers.manifest.manifest_v1 import ManifestV1
+from dbt_artifacts_parser.parsers.manifest.manifest_v2 import ManifestV2
+from dbt_artifacts_parser.parsers.manifest.manifest_v3 import ManifestV3
+from dbt_artifacts_parser.parsers.manifest.manifest_v4 import ManifestV4
+from dbt_artifacts_parser.parsers.manifest.manifest_v5 import ManifestV5
+from dbt_artifacts_parser.parsers.manifest.manifest_v6 import ManifestV6
+from dbt_artifacts_parser.parsers.manifest.manifest_v7 import ManifestV7
+from dbt_artifacts_parser.parsers.manifest.manifest_v8 import ManifestV8
+from dbt_artifacts_parser.parsers.manifest.manifest_v9 import ManifestV9
+from dbt_artifacts_parser.parsers.manifest.manifest_v10 import ManifestV10
+from dbt_artifacts_parser.parsers.manifest.manifest_v11 import ManifestV11
+
+Manifest = Union[
```

Comment:
I think it would be best if dbt_artifacts_parser provided these types :)

Reply:
@syou6162 hmm maybe good to make a pull request to dbt_artifacts_parser?!
I'm just concerning that this `Union` of all versions, that will get burden of dbterd side which will require a change to align with dbt_artifacts_parser. This is also a reason why we [relax its version here](https://github.com/datnguye/dbterd/blob/main/pyproject.toml#L32).

Do you have any ideas to avoid that? Thanks

Reply:
@datnguye I think your concern is valid.

OK, I understand. I will try to send a pull request to dbt-artifacts-parser to provide these types. However, it may take some time for the pull request to be reviewed or incorporated into the new version. In my experience, it often takes a long time to be reviewed. By the way, this repository is very helpful for me as a contributor because reviews are done quickly.

When a third party like me is involved in the development, it is a big help to have types in the arguments and return values of functions using mypy. So how about proceeding in this way?

- Type definition using `Union` in dbterd (this pull request)
- Send a pull request to dbt-artifacts-parser to add the `Union` type definition, and it will be included in the new version.
- Replace the types defined in dbterd with the types defined in dbt-artifacts-parser

If you have difficulty accepting this proposal, I will add a new commit to remove the type definitions for `Manifest` and `Catalog` for now. However, it is the type definitions for `Manifest` and `Catalog` that I originally mainly wanted to add 😭

Reply:
@datnguye I have done some investigation and will add a note. It is possible to get the return type of a function via mypy, but it is not a good idea since it is displayed as `Any` on IDEs like VSCode, and it is not easy to get to the type definition. It seems that this type information can only be obtained at runtime.

```python
from typing import get_type_hints, TypeAlias
from dbt_artifacts_parser.parser import parse_manifest

Manifest: TypeAlias = get_type_hints(parse_manifest).get("return")
tmp: Manifest
```

<img width="619" alt="スクリーンショット 2024-02-11 1 22 00" src="https://github.com/datnguye/dbterd/assets/18356/b5badb35-ee66-4ee2-bbdc-4b2af9276507">

Reply:
Thanks @syou6162 I think I will be convinced by current impl, but let me take a look further until Tuesday. If we can not see any alternatives, lets merge it right away and tag v1.11.0b2.

The major 1.11.0 should be available by end of next week 👍 🏃

Reply:
Just FYI, I have submitted a pull request to dbt-artifacts-parser to provide a type using `Union`.

- https://github.com/yu-iskw/dbt-artifacts-parser/pull/73